src/Uno.UI.Runtime.Skia.Wpf/Extensions/ApplicationModel/DataTransfer/ClipboardExtensions.cs

```diff
@@ -0,0 +1,231 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.IO;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Interop;
+using System.Windows.Media.Imaging;
+using Uno.ApplicationModel.DataTransfer;
+using Uno.UI.Skia.Platform;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.Storage;
+using Windows.Storage.Streams;
+using Windows.UI.Core;
+using Clipboard = System.Windows.Clipboard;
+
+namespace Uno.Extensions.ApplicationModel.DataTransfer
+{
+	internal class ClipboardExtensions : IClipboardExtension
+	{
+		const int WM_CLIPBOARDUPDATE = 0x031D;
+
+		private readonly WpfHost _host;
+		private HwndSource _hwndSource;
+		private bool _pendingStartContentChanged;
+
+		public event EventHandler<object> ContentChanged;
+
+		public ClipboardExtensions(object owner)
+		{
+			_host = WpfHost.Current;
+
+			// This class may be accessed before the Window is loaded
+			// if the Clipboard is somehow accessed really early.
+			if (_host.IsLoaded)
+			{
+				HostLoaded(null, null);
+			}
+			else
+			{
+				// Hook for native events
+				_host.Loaded += HostLoaded;
+			}
+
+			void HostLoaded(object sender, RoutedEventArgs e)
+			{
+				_host.Loaded -= HostLoaded;
+
+				var win = Window.GetWindow(_host);
+
+				var fromDependencyObject = PresentationSource.FromDependencyObject(win);
+				_hwndSource = fromDependencyObject as HwndSource;
+
+				if (_pendingStartContentChanged)
+                {
+					StartContentChanged();
+                }
+			}
+		}
+
+		public void StartContentChanged()
+		{
+			if (_hwndSource != null)
+			{
+				_hwndSource.AddHook(OnWmMessage);
+				ClipboardNativeFunctions.AddClipboardFormatListener(_hwndSource.Handle);
+			}
+			else
+            {
```

Comment:
And here the indentation seems also distorted 😀

Reply:
Probably the consequences of copy-paste from my experimental app to the library. Sorry for that.
(And why doesn't Codacy catch it?)